forem/forem
# Member index (Phase 2) - Invitations actions

## What type of PR is this? (check all applicable)


## Description

This PR adds the actions menu to the invitations table, along with the "Resend invite" and "Cancel invite" actions. It also applies some styles to the overall table.

## Related Tickets & Documents



- Closes https://github.com/forem/forem/issues/17214
- Closes https://github.com/forem/forem/issues/17215

## QA Instructions, Screenshots, Recordings

- Make sure you have the `member_index_view` feature flag enabled, then visit `/admin/invitations`
- If you don't have any invited users in the table, invite some either via the "Invite member" button, or in the console (`User.invite!(email: "someone@something.com", name: "Test user", username: "testuser", registered: false)`)
- Each invited member should have an overflow menu with the "Resend invite" and "Cancel invite" options
- Clicking to resend invite should re-invite the member, and a confirmation message should appear at the top of the page
- Clicking to cancel an invite should delete the invitation, removing it from the table, and displaying a confirmation message at the top of the page


https://user-images.githubusercontent.com/20773163/162940467-24089321-7544-4f9a-888d-0e13080f4854.mp4


### UI accessibility concerns?

- Each dropdown menu has an `aria-label` tailored to the individual invite (e.g. "Invitation actions: Test user")
- Dropdowns are powered by the same code as in the main member index view, considering accessible focus patterns
- The buttons inside the dropdown are individual form submits, so a screen reader user may have some context that clicking this button is going to submit a change and reload the page

## Added/updated tests?



## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] This change does not need to be communicated, and this is why not: part of a larger issue behind a feature flag for now


## [optional] What gif best describes this PR or how it makes you feel?

![calling action!](https://media.giphy.com/media/EukOAWKKs0QoW3ps7Z/giphy.gif)
